bundles/org.openhab.binding.sedif/src/main/java/org/openhab/binding/sedif/internal/api/SedifHttpApi.java

```diff
@@ -0,0 +1,377 @@
+                Fields fields = new Fields();
+
+                request = request.method(HttpMethod.POST);
+
+                AuraContext appCtx = bridgeHandler.getAppCtx();
+                if (appCtx != null) {
+                    fields.put("aura.context", getAuraContextPayload(appCtx));
+                }
+
+                String token = bridgeHandler.getToken();
+                fields.put("aura.token", token);
+
+                if (handler != null) {
+                    String contractId = handler.getContractId();
+                    String meterIdB = handler.getMeterIdB();
+                    String meterIdA = handler.getMeterIdA();
+
+                    if (!"".equals(contractId)) {
+                        paramsSub.put("contratId", contractId);
+                        paramsSub.put("contractId", contractId);
+                    }
+                    if (!"".equals(meterIdB)) {
+                        paramsSub.put("NUMERO_COMPTEUR", meterIdB);
+                    }
+
+                    if (!"".equals(meterIdA)) {
+                        paramsSub.put("ID_PDS", meterIdA);
+                    }
+                }
+
+                String msg = getActionPayload(cmd);
+                fields.put("message", msg);
+
+                request = request.content(new FormContentProvider(fields));
+            }
+
+            ContentResponse result = request.send();
+            if (result.getStatus() == HttpStatus.TEMPORARY_REDIRECT_307
+                    || result.getStatus() == HttpStatus.MOVED_TEMPORARILY_302) {
+                String loc = result.getHeaders().get("Location");
+                String newUrl = "";
+
+                if (loc.startsWith("http://") || loc.startsWith("https://")) {
+                    newUrl = loc;
+                } else {
+                    newUrl = bridgeHandler.getBaseUrl() + loc.substring(1);
+                }
+
+                request = httpClient.newRequest(newUrl);
+                request = request.method(HttpMethod.GET);
+                result = request.send();
+
+                if (result.getStatus() == HttpStatus.TEMPORARY_REDIRECT_307
+                        || result.getStatus() == HttpStatus.MOVED_TEMPORARILY_302) {
+                    loc = result.getHeaders().get("Location");
+                    String[] urlParts = loc.split("/");
+                    if (urlParts.length < 4) {
+                        throw new SedifException("malformed url : %s", loc);
+                    }
+                    return urlParts[3];
+                }
+            }
+            if (result.getStatus() != 200) {
+                throw new SedifException("Error requesting '%s': %s", url, result.getContentAsString());
+            }
+
+            String content = result.getContentAsString();
+            return content;
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            throw new SedifException(e, "Error getting url: '%s'", url);
```

Comment:
interrupt exception should be handled different. The isinterrupted flag should be enabled again.